### PR TITLE
EDSC-4055: Add performance based testing to our code base to assist prevent regression

### PR DESCRIPTION
**Pull request by Bccorb**
# Overview

### What is the feature?

Adding some basic metrics for tracking in Playwright tests. LCP tracking and Time to Last Byte of the Search pages load time.

### What is the Solution?

Added two playwright tests to capture the LCP and the response end time of the search page loading. This correlates to the time it takes to receive all javascript code.

### What areas of the application does this impact?

No impact to the actual application.

# Testing

### Reproduction steps

Running the playwright tests will show allow for inspection of the metrics.

### Attachments
Search page metrics
Screenshot 2024-06-03 at 10 23 39 AM

LCP passing test. 
Screenshot 2024-06-03 at 10 24 19 AM

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: tests/e2e/performace/performance.spec.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    await page.route('**/*.{png,jpg,jpeg}', (route) => route.abort())
+    await page.route(/collections/, async (route) => {
+      await route.fulfill({
+        json: singleCollection.body,
+        headers: singleCollection.headers
+      })
+    })
+  })
+
+  test('Search page load time is less than 1 second', async ({ page, browserName }) => {
```
Comment (eudoroolivares2016) [inline tests/e2e/performace/performance.spec.js:16]: We should update the test string since it looks like we are comparing to 3 seconds here

**Review by eudoroolivares2016 — APPROVED** [review]

File: tests/e2e/performace/performance.spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      expect(request.timing().responseEnd < 30000).toBe(true)
+    }
+  })
+
+  test('Search page LCP start time is less than 7 second', async ({ page, browserName }) => {
+    if (browserName === 'chromium') {
+      await page.goto('/')
+      page.on('metrics', (metrics) => {
+        const lcp = metrics.LargestContenfulPaint / 1000
+        if (lcp > 7) {
```
Comment (eudoroolivares2016) [inline tests/e2e/performace/performance.spec.js:31]:
```suggestion
        expect(lcp).toBeLessThan(7)
```

**Review by eudoroolivares2016 — DISMISSED** [review]
Minor comments

File: tests/e2e/performace/performance.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    })
+  })
+
+  test('Search page load time is less than 1 second', async ({ page, browserName }) => {
+    if (browserName === 'chromium') {
+      await page.goto('/')
+      const requestFinishedPromise = page.waitForEvent('requestfinished')
+      const request = await requestFinishedPromise
+
+      expect(request.timing().responseEnd < 30000).toBe(true)
```
Comment (dpesall) [inline tests/e2e/performace/performance.spec.js:22]: Is this supposed to be `30000`ms? This is 30 seconds.

File: tests/e2e/performace/performance.spec.js (hunk 1 above)
Comment (Bccorb) [reply tests/e2e/performace/performance.spec.js:16]: Updated.

**Review by rushgeo — APPROVED** [review]

**Review by eudoroolivares2016 — APPROVED** [review]

**Pull request closed without merging**